subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputFilePropertiesIntegrationTest.groovy

```diff
@@ -68,4 +68,38 @@ class TaskInputFilePropertiesIntegrationTest extends AbstractIntegrationSpec {
         where:
         method << ["file", "dir"]
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/3792")
+    def "task dependency is discovered via Buildable input files"() {
+        buildFile << """
+            @groovy.transform.TupleConstructor
+            class BuildableArtifact implements Buildable, Iterable<File> {
+                FileCollection files
+
+                Iterator<File> iterator() {
+                    files.iterator()
+                }
+
+                TaskDependency getBuildDependencies() {
+                    files.getBuildDependencies()
+                }
+            }
+
+            task foo {
+                outputs.file "foo.txt"
+                doFirst {}
+            }
+
+            task bar {
+                inputs.files(new BuildableArtifact(foo.outputs.files))
```

Comment:
Don't we want to deprecate `outputs.getFiles()`, too?

Reply:
I'm not completely convinced that we should, but in any case I changed the code now so that it works without referring to outputs directly.